### PR TITLE
chore: bump node-driver-registrar version

### PR DESCRIPTION
**Pull request by Devinwong**
**What type of PR is this?**
/kind chore

**What this PR does / why we need it**:
bump node-driver-registrar version
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/components.json [hunk 1]
```diff
@@ -416,11 +416,11 @@
       "multiArchVersionsV2": [
         {
           "renovateTag": "registry=https://mcr.microsoft.com, name=oss/kubernetes-csi/csi-node-driver-registrar",
-          "latestVersion": "v2.11.1"
+          "latestVersion": "v2.12.0"
         },
         {
           "renovateTag": "registry=https://mcr.microsoft.com, name=oss/kubernetes-csi/csi-node-driver-registrar",
-          "latestVersion": "v2.10.1"
+          "latestVersion": "v2.11.1"
```
[the remark this answers is not in the document]
Comment (andyzhangx) [reply parts/linux/cloud-init/artifacts/components.json:423]: both `v2.11.1` and `v2.10.1` are used in aks-rp, we need to keep them

```
  {{- if semverCompare ">=1.30.0" .version -}}v2.11.1
  {{- else if semverCompare ">=1.27.0" .version -}}v2.10.1
```

[the remark this answers is not in the document]
Comment (andyzhangx) [reply parts/linux/cloud-init/artifacts/components.json:423]: windows image should also be bumped

[the remark this answers is not in the document]
Comment (andyzhangx) [reply parts/linux/cloud-init/artifacts/components.json:423]: I have worked out a PR to update registrar and liveness-probe images in both linux and windows vhd: https://github.com/Azure/AgentBaker/pull/5073

[the remark this answers is not in the document]
Comment (Devinwong) [reply parts/linux/cloud-init/artifacts/components.json:423]: okay I will abandon this PR then.

**Pull request closed without merging**